libs/partners/nvidia-ai-endpoints/langchain_nvidia_ai_endpoints/chat_models.py

```diff
@@ -229,7 +238,70 @@ def preprocess_msg(self, msg: BaseMessage) -> Dict[str, str]:
     def custom_postprocess(self, msg: dict) -> str:
         if "content" in msg:
             return msg["content"]
-        logger.warning(
-            f"Got ambiguous message in postprocessing; returning as-is: msg = {msg}"
-        )
+        elif "b64_json" in msg:
+            return msg["b64_json"]
         return str(msg)
+
+    ######################################################################################
+    ## Core client-side interfaces
+
+    def get_generation(
+        self,
+        inputs: Sequence[Dict],
+        **kwargs: Any,
+    ) -> dict:
+        """Call to client generate method with call scope"""
+        stop = kwargs.get("stop", None)
+        payload = self.get_payload(inputs=inputs, stream=False, **kwargs)
+        out = self.client.get_req_generation(self.model, stop=stop, payload=payload)
+        return out
+
+    def get_stream(
+        self,
+        inputs: Sequence[Dict],
+        **kwargs: Any,
+    ) -> Iterator:
+        """Call to client stream method with call scope"""
+        stop = kwargs.get("stop", None)
+        payload = self.get_payload(inputs=inputs, stream=True, **kwargs)
+        return self.client.get_req_stream(self.model, stop=stop, payload=payload)
+
+    def get_astream(
+        self,
+        inputs: Sequence[Dict],
+        **kwargs: Any,
+    ) -> AsyncIterator:
+        """Call to client astream methods with call scope"""
+        stop = kwargs.get("stop", None)
+        payload = self.get_payload(inputs=inputs, stream=True, **kwargs)
+        return self.client.get_req_astream(self.model, stop=stop, payload=payload)
+
+    def get_payload(self, inputs: Sequence[Dict], **kwargs: Any) -> dict:
+        """Generates payload for the _NVIDIAClient API to send to service."""
+        const_kwargs = {k: getattr(self, k) for k in self.model_kws}
```

Comment:
```suggestion
        const_kwargs = {
            "temperature": self.temperature,
            "max_tokens": self.max_tokens,
            "top_p": self.top_p,
            "seed": self.seed,
            "bad": self.bad,
            "stop": self.stop,
            "labels": self.labels,
        }
```

Reply:
Also `const` might be the wrong prefix. Maybe `attr_kwargs`?

Reply:
Oh, I see what you're asking...